src/main/java/com/mraof/minestuck/item/IncompleteSburbCodeItem.java

```diff
@@ -0,0 +1,176 @@
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.HitResult;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * Used for the Sburb Code item, extends ReadableSburbCodeItem which is used by Completed Sburb Code.
+ * Allows players to store a list of blocks from the GREEN_HIEROGLPYH block tag that in game reflects the collection of genetic code for the creation of sburb
+ */
+public class IncompleteSburbCodeItem extends ReadableSburbCodeItem
+{
+	public IncompleteSburbCodeItem(Properties properties)
+	{
+		super(properties);
+	}
+	
+	@Override
+	public InteractionResult useOn(UseOnContext context)
+	{
+		ItemStack stackInUse = context.getItemInHand();
+		Level worldIn = context.getLevel();
+		Player player = context.getPlayer();
+		InteractionHand handIn = context.getHand();
+		
+		if(!worldIn.isClientSide && player != null)
+		{
+			BlockHitResult blockRayTraceResult = BlockHitResultUtil.getPlayerPOVHitResult(worldIn, player);
+			
+			if(blockRayTraceResult.getType() == HitResult.Type.BLOCK)
+			{
+				BlockState raytraceState = BlockHitResultUtil.collidedBlockState(player, blockRayTraceResult);
+				
+				List<Block> preProcessBlockList = getRecordedBlocks(stackInUse); //used to differentiate from after processing
+				ItemStack processedStack = addCarvingsToCode(raytraceState.getBlock(), stackInUse); //the call to addCarvingsToCode is what processes the stack
+				List<Block> postProcessBlockList = getRecordedBlocks(processedStack);
+				if(!preProcessBlockList.containsAll(postProcessBlockList))
+				{
+					worldIn.playSound(null, player.blockPosition(), SoundEvents.VILLAGER_WORK_CARTOGRAPHER, SoundSource.BLOCKS, 1.0F, 1.0F);
+				}
+			}
+			
+			attemptConversionToCompleted(player, handIn); //if after addCarvingsToCode the item now possesses all hieroglyphs, convert it to a completed sburb code item
+		}
+		
+		return InteractionResult.PASS;
+	}
+	
+	/**
+	 * Checks if the block being investigated is in the GREEN_HIEROGLYPHS block tag, if true then it passes to addRecordedInfo()
+	 */
+	public ItemStack addCarvingsToCode(Block hieroglyphBlock, ItemStack stackInUse)
+	{
+		List<Block> hieroglpyhsList = MSTags.getBlocksFromTag(MSTags.Blocks.GREEN_HIEROGLYPHS);
+		
+		if(hieroglpyhsList.contains(hieroglyphBlock))
+		{
+			for(Block block : hieroglpyhsList)
+			{
+				if(block == hieroglyphBlock)
+				{
+					addRecordedInfo(stackInUse, hieroglyphBlock);
+					break;
+				}
+			}
+		}
```

Comment:
Looks like you're checking that `hieroglyphsList` contains `hieroglyphBlock` twice; once with a function, and then once manually.

Reply:
Removed the if statement

Reply:
Rather than removing `contains()`, how about keeping specifically that part:
```java
		if(hieroglpyhsList.contains(hieroglyphBlock)
		{
			addRecordedInfo(stackInUse, hieroglyphBlock);
		}
```